src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,174 @@
 package core.basesyntax;
 
+import java.util.Collection;
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> currentLastNode = tail;
+        final Node<T> newNode = new Node<>(currentLastNode, value, null);
+        tail = newNode;
+
+        if (currentLastNode == null) {
+            head = newNode;
+        } else {
+            currentLastNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            linkBefore(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        addAll(size, list);
+    }
+
+    private void addAll(int index, Collection<? extends T> collection) {
+        checkAddingIndex(index);
+        T[] objects = (T[]) collection.toArray();
+
+        if (index == size) {
+            for (T element : objects) {
+                add(element);
+            }
+        } else {
+            for (T element : objects) {
+                linkBefore(element, getNode(index));
+                index++;
+            }
+        }
```

Comment:
What do you think about this implementation? I did not test it, but hope it works)
![image](https://user-images.githubusercontent.com/20326858/216312948-98d7929d-6ef6-430f-91cd-80237995185a.png)

Reply:
> What do you think about this implementation? I did not test it, but hope it works) ![image](https://user-images.githubusercontent.com/20326858/216312948-98d7929d-6ef6-430f-91cd-80237995185a.png)

fixed)

Reply:
not fixed